UnityProject/Assets/Scripts/Machines/Jukebox.cs

```diff
@@ -0,0 +1,304 @@
+		if (integrity.integrity > integrity.initialIntegrity / 2)
+		{
+			IsPlaying = true;
+			spriteHandler.SetSpriteSO(SpritePlaying);
+
+			AudioSourceParameters audioSourceParameters = new AudioSourceParameters
+			{
+				MixerType = MixerType.Muffled,
+				SpatialBlend = 1, // 3D, we need it to attenuate with distance
+				Volume = Volume,
+				MinDistance = MinSoundDistance,
+				MaxDistance = MaxSoundDistance,
+				VolumeRolloffType = VolumeRolloffType.EaseInAndOut,
+				Spread = Spread
+
+			};
+
+			AddressableAudioSource addressableAudioSourceFromCache = await GetAddressableAudioSourceFromCache();
+			soundSpawnToken = await SoundManager.PlayNetworkedAtPos(addressableAudioSourceFromCache, registerTile.WorldPositionServer, audioSourceParameters, false, true, gameObject);
+
+			startPlayTime = Time.time;
+			UpdateGUIAsync();
+		}
+	}
+
+	public void Stop()
+	{
+		IsPlaying = false;
+
+		if (integrity.integrity >= integrity.initialIntegrity / 2)
+			spriteHandler.SetSpriteSO(SpriteIdle);
+		else
+			spriteHandler.SetSpriteSO(SpriteDamaged);
+
+		SoundManager.StopNetworked(soundSpawnToken);
+
+		UpdateGUIAsync();
+	}
+
+	public void PreviousSong()
+	{
+		if (currentSongTrackIndex > 0)
+		{
+			if (IsPlaying)
+				SoundManager.StopNetworked(soundSpawnToken);
+
+			// JESTER
+			// We unload the music from the library, freeing RAM
+			//SoundManager.Instance.UnloadMusic(SoundManager.Instance.MusicLibrary.ElementAt(currentSongTrackIndex).Path);
+
+			currentSongTrackIndex--;
+			UpdateGUIAsync();
+
+			if (IsPlaying)
+				Play();
+		}
+	}
+
+	public bool NextSong()
+	{
+		if (currentSongTrackIndex < SoundManager.Instance.MusicLibrary.Count - 1)
+		{
+			if (IsPlaying)
+				SoundManager.StopNetworked(soundSpawnToken);
+
+			// JESTER
+			// We unload the music from the library, freeing RAM
+			// SoundManager.Instance.UnloadMusic(SoundManager.Instance.MusicLibrary.ElementAt(currentSongTrackIndex).Path);
+
+			currentSongTrackIndex++;
+			UpdateGUIAsync();
+
+			if (IsPlaying)
+				Play();
+
+			return true;
+		}
+		else
+			return false;
+	}
+
+	public void VolumeChange(float newVolume)
+	{
+		Volume = newVolume;
+
+		AudioSourceParameters audioSourceParameters = new AudioSourceParameters
+		{
+			Volume = newVolume,
+		};
+
+		ChangeAudioSourceParametersMessage.SendToAll(soundSpawnToken, audioSourceParameters);
+	}
+
+	private void OnDamageReceived(DamageInfo damageInfo)
+	{
+		if (integrity.integrity <= integrity.initialIntegrity / 2)
+		{
+			Stop();
+		}
+	}
+
+	private async void UpdateGUIAsync()
+	{
+		List<ElementValue> valuesToSend = new List<ElementValue>();
+		valuesToSend.Add(new ElementValue() { Id = "TextTrack", Value = Encoding.UTF8.GetBytes(TrackPosition) });
+		valuesToSend.Add(new ElementValue() { Id = "TextSong", Value = Encoding.UTF8.GetBytes(await GetSongNameAsync()) });
+		valuesToSend.Add(new ElementValue() { Id = "TextArtist", Value = Encoding.UTF8.GetBytes(await GetArtistNameAsync()) });
```

Comment:
Codacy found an issue: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)